test/presubmit-tests-with-pipeline-deployment.sh

```diff
@@ -63,6 +63,12 @@ while [ "$1" != "" ]; do
     shift
 done
 
+# Making the Prow's PULL_PULL_SHA variable more reliable:
+# PULL_PULL_SHA is empty when Pros/Tide tests the batches.
+# PULL_BASE_SHA cannot be used here as it still points to master tip in that case.
+
+export PULL_PULL_SHA=$(git rev-parse HEAD)
```

Comment:
I think this is a good fix for batch test, but it changes semantics for presubmit test.

For example, your commit is 2a97060ccc690d2522d8a22140cb16c3bbbc158e, but when you look at your test log, it says ca4d02ce7c0ece3b455677936f1e99ba454b6950. That's because when running presubmit test, it first merges with master before running the test.

[log](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubeflow_pipelines/1958/kubeflow-pipeline-e2e-test/1166142311336251392#0:build-log.txt%3A18)

I don't know better ways to get a PULL_PULL_SHA. Maybe it's good enough to fill this variable only when it's empty.